src/main/java/calculator/Application.java

```diff
@@ -1,7 +1,157 @@
 package calculator;
 
+import camp.nextstep.edu.missionutils.Console;
+
+/**
+ * 이 클래스는 문자열 덧셈 계산기를 나타낸다.
+ * 커스텀 구분자가 있는지 확인하는 메서드, 구분자 배열을 반환하는 메서드,
+ * 커스텀 구분자 입력 부분을 제거하는 메서드, 구분자를 콤마로 대체하는 메서드,
+ * 문자열을 구분자를 기준으로 나누는 메서드, 입력값의 유효성을 검증하는 메서드,
+ * 배열의 합을 반환하는 메서드로 구성된다.
+ */
 public class Application {
+    /** 입력 문자열에 커스텀 구분자가 있는지 확인하여 bool 값을 반환한다.
+     * 여기서 커스텀 구분자란 입력 문자열 앞 부분의 "//"와 "\n" 사이에 위치하는 문자를 말한다.
+     *
+     * @param input 입력 문자열
+     * @return 커스텀 구분자 존재 여부
+     */
+    public static boolean hasCustomSeparator(String input) {
+//        if (input.startsWith("//\\n")) { // empty separator 비허용 시
+//            return false;
+//        }
+        return input.startsWith("//") && input.contains("\\n");
+    }
+
+    /**
+     * 입력 문자열에서 ","를 제외한 구분자를 배열로 반환한다.
+     * 즉, 커스텀 구분자가 없다면 ":"만, 있다면 ":"와 해당 구분자를 포함한 배열을 반환한다.
+     *
+     * @param input 입력 문자열
+     * @return 구분자의 배열
+     */
+    public static String[] getSeparators(String input) {
+        String[] separators = new String[]
+                {":"};
+        if (hasCustomSeparator(input)) {
+            separators =
+                    new String[]{":",
+                            input.substring(2,
+                                    input.indexOf("\\n"))};
+        }
+        return separators;
+    }
+
+    /**
+     * 입력 문자열에서 커스텀 구분자 처리 부분을 제외하고 반환한다.
+     * 즉, 커스텀 구분자가 있다면 입력 문자열의 "\n" 이후만,
+     * 없다면 입력 문자열을 그대로 반환한다.
+     *
+     * @param input 입력 문자열
+     * @return 잘린 입력 문자열
+     */
+    public static String trimCustomSeparator(String input) {
+        String trimedInput = input;
+        if (hasCustomSeparator(input)) {
+            trimedInput = input.substring(input.indexOf("\\n") + 2);
+        }
+        return trimedInput;
+    }
+
+    /**
+     * 문자열에서 ","를 제외한 구분자를 모두 ","로 대체한다.
+     *
+     * @param input 입력 문자열
+     * @param separators 구분자 배열
+     * @return 대체된 입력 문자열
+     */
+    public static String replaceSeparators(String input, String[] separators) {
+        String replacedInput = trimCustomSeparator(input);
+        for (String sep : separators) {
+            replacedInput = replacedInput.replace(sep, ",");
+            if (sep.isEmpty()) {//empty separator 고려
```

Comment:
```suggestion
if(!sep.isEmpty()) continue;

```
로 처리하면 밑에 쪽의 코드 블록이 한단 줄어듭니다.

코드 블록 들여쓰는 곳이 많은데 충분히 완화할 수 있을 것이라고 생각합니다

Reply:
주 언어가 파이썬이다보니 들여쓰는 데에 익숙해져 있네요. 앞으로 당겨두는 게 가독성에 좀 더 나을까요?